src/cryptographic_primitives/hashing/ext.rs

```diff
@@ -143,9 +142,9 @@ pub trait HmacExt: Sized {
     fn verify_bigint(self, code: &BigInt) -> Result<(), MacError>;
 }
 
-impl<D> HmacExt for Hmac<D>
+impl<D> MacExt for D
 where
-    D: digest::Input + digest::BlockInput + digest::FixedOutput + digest::Reset + Default + Clone,
+    D: Mac,
 {
     fn new_bigint(key: &BigInt) -> Self {
```

Comment:
Note that this function contains an unwrap (line [152]). It was safe as HMAC is expected to handle any key length, but it's not true for generic MAC implementation. Basically, that's why I extended only hmac - otherwise API is changed and a lot of place need to be updated

[152]: https://github.com/ZenGo-X/curv/pull/129/files#diff-b4d0e9a8c5d053ef1014eb99ea89f12058b1b11525985d68063aa67140645203L152

Reply:
I reverted this change, good catch!
too bad there's no easy way to make this abstract over any Mac